src/pages/settings/panes/Appearance.tsx

```diff
@@ -372,7 +372,16 @@ export function Component(props: Props) {
                                     }
                                 />
                             </div>
-                            <span style={`color: ${getContrastingColour(theme[x])}`}>{x}</span>
+                            <span
+                                style={`color: ${
+                                    theme[x].toUpperCase() === "TRANSPARENT"
+                                        ? getContrastingColour(
+                                              theme["primary-background"],
+                                          )
+                                        : getContrastingColour(theme[x])
```

Comment:
Instead of explicitly checking for it to be transparent, we should fallback to the contrast background colour if `getConstrastingColour` fails to parse, this will also account for invalid data.